src/anyio/abc/_synchronization.py

```diff
@@ -1,10 +1,37 @@
 from abc import ABCMeta, abstractmethod
+from dataclasses import dataclass
 from types import TracebackType
 from typing import Optional, Type, TypeVar
 
 T_Retval = TypeVar('T_Retval')
 
 
+@dataclass(frozen=True)
+class EventStatistics:
+    """
+    :ivar int tasks_waiting: number of tasks waiting on :meth:`~.Event.wait`
+    """
+
+    tasks_waiting: int
+
+
+@dataclass(frozen=True)
+class CapacityLimiterStatistics:
+    """
+    :ivar int borrowed_tokens: number of tokens currently borrowed by tasks
+    :ivar float total_tokens: total number of available tokens
+    :ivar tuple borrowers: tasks or other objects currently holding tokens borrowed from this
+        limiter
+    :ivar int tasks_waiting: number of tasks waiting on :meth:`~.CapacityLimiter.acquire` or
+        :meth:`~.CapacityLimiter.acquire_on_behalf_of`
+    """
+
+    borrowed_tokens: int
+    total_tokens: float
+    borrowers: tuple
```

Comment:
```suggestion
    borrowers: Tuple[object]
```

Reply:
Why? Borrowers can be any type, but this would mean only `object()` instances can go to `borrowers`.

Reply:
every instance is an instance of object

Reply:
PEP 484 is pretty clear on this. A plain `tuple` which I had here, is equivalent to `Tuple[Any]` which is NOT the same as `Tuple[object]`.

Reply:
Reading up on it once more, it's not so *clear* why this is so, but the `Any` type was introduced specifically for this purpose.

Reply:
sorry I meant:

```suggestion
    borrowers: Tuple[object, ...]
```

Reply:
yes I know, using `tuple`, and alias of `Tuple[Any, ...]` mypy considers the following sound:

```python
capacity_limiter = anyio.create_capacity_limiter()
capacity_limiter.acquire_on_behalf_of(current_task())
borrowers: Tuple[int, ...] = capacity_limiter.statistics().borrowers
v: int = borrowers[0]  # woops v is actually a Task!
```

Reply:
Yeah but `Tuple[Any, ...]` is what we want and (I correct myself) `tuple` is equivalent to `Tuple[Any, ...]`.
The difference between `object` and `Any` comes down to type substitution rules which are not relevant here.
Bottom line is, `tuple` is correct.